src/batched/KokkosBatched_QR_WithColumnPivoting_TeamVector_Internal.hpp

```diff
@@ -158,7 +159,7 @@ namespace KokkosBatched {
 	  if (m_atl == 0) max_diag = ats::abs(A[0]);
 	  const value_type
 	    val_diag = ats::abs(A_part3x3.A11[0]),
-	    threshold(max_diag*ats::epsilon());
+	    threshold(10*max_diag*ats::epsilon());
```

Comment:
You can verify using a tall skinny matrix of all ones that epsilon*max_diag is likely exceeded by what sits on the diagonal of R (by a very small amount). Basically, this is an accumulation of machine precision errors.